### PR TITLE
[T2A2][T09-A3] Tang Di Feng

### PR DESCRIPTION
**Pull request by e0011840**

### CODE REVIEW

**Comment by e0011840** [comment]
Ready for review

File: src/seedu/addressbook/AddressBook.java [hunk 1]
```diff
@@ -394,10 +394,22 @@ private static String executeCommand(String userInputString) {
      * @return  size 2 array; first element is the command type and second element is the arguments string
      */
     private static String[] splitCommandWordAndArgs(String rawUserInput) {
-        final String[] split =  rawUserInput.trim().split("\\s+", 2);
-        return split.length == 2 ? split : new String[] { split[0] , "" }; // else case: no parameters
+        final String[] splitCommandWordandArgs =  trimRawUserInput(rawUserInput);
+        if(splitCommandWordandArgs.length == 2) {
+        	return splitCommandWordandArgs;
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:410]: Notice that your indentation mechanism uses tabs for indentation. 
The coding standards we follow require using 4 spaces for indentation.
Tip: you can set Eclipse to do this for you, check out how!

File: src/seedu/addressbook/AddressBook.java [hunk 2]
```diff
@@ -1158,12 +1158,12 @@ private static String getUsageInfoForExitCommand() {
...
-     * @param s  Parameter as a string
-     * @param sign  Parameter sign to be removed
+     * @param fullString  Parameter as a string
+     * @param prefix  Parameter sign to be removed
      * @return  string without the sign
      */
-    private static String removePrefixSign(String s, String sign) {
-        return s.replace(sign, "");
+    private static String removePrefix(String fullString, String prefix) {
+        return fullString.replace(prefix, "");
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:1177]: There are a couple of things you need to take care with this commit.
1. Notice that you haven't updated the method to remove the prefix "the correct way"; your solution is still the same as the given approach. Please read the description carefully.
2. When you are abstracting the methods to make them generic, the header comment should reflect the same; your header comment is still specific to address book due to reference to p/ d/ etc.,

File: src/seedu/addressbook/AddressBook.java [hunk 3]
```diff
@@ -208,7 +208,18 @@
 
     public static void main(String[] args) {
         showWelcomeMessage();
-        processProgramArgs(args);
+        if (args.length >= 2) {
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:211]: This commit is unrelated to T2A2.
Tip: Avoid accumulating commits in your pull request by branching properly!

**Review by okkhoy — COMMENTED** [review]
@e0011840 some comments added. please take a look; ack and close the pull request.
On an unrelated note, please add a clean photo of yourself on the github profile so that I can identify who has submitted the PR.

**Comment by e0011840** [comment]
Okay Thank You! Noted!!
